packages/core/src/Stores/notification-store.js

```diff
@@ -0,0 +1,897 @@
+    }
+
+    @action.bound
+    addNotificationBar(message) {
+        this.push_notifications.push(message);
+        this.push_notifications = unique(this.push_notifications, 'msg_type');
+    }
+
+    @action.bound
+    addNotificationMessage(notification) {
+        if (!notification) return;
+        if (!this.notification_messages.find(item => item.key === notification.key)) {
+            // Remove notification messages if it was already closed by user and exists in LocalStore
+            const active_loginid = LocalStore.get('active_loginid');
+            const messages = LocalStore.getObject('notification_messages');
+
+            if (active_loginid) {
+                // Check if is existing message to remove already closed messages stored in LocalStore
+                const is_existing_message = Array.isArray(messages[active_loginid])
+                    ? messages[active_loginid].includes(notification.key)
+                    : false;
+
+                if (is_existing_message) {
+                    this.markNotificationMessage({ key: notification.key });
+                }
+
+                const sortFn = isMobile() ? sortNotificationsMobile : sortNotifications;
+                this.notification_messages = [...this.notification_messages, notification].sort(sortFn);
+
+                if (!excluded_notifications.includes(notification.key)) {
+                    this.updateNotifications(this.notification_messages);
+                }
+            }
+        }
+    }
+
+    @action.bound
+    addNotificationMessageByKey(key) {
+        if (key) this.addNotificationMessage(this.client_notifications[key]);
+    }
+
+    @action.bound
+    addVerificationNotifications(identity, document) {
+        if (identity.status === 'expired') this.addNotificationMessage(this.client_notifications.poi_expired);
+
+        if (document.status === 'expired') this.addNotificationMessage(this.client_notifications.poa_expired);
+    }
+
+    @action.bound
+    filterNotificationMessages() {
+        if (LocalStore.get('active_loginid') !== 'null')
+            this.resetVirtualBalanceNotification(LocalStore.get('active_loginid'));
+        this.notifications = this.notification_messages.filter(notification => {
+            if (notification.platform === undefined || notification.platform.includes(getPathname())) {
+                return true;
+            } else if (!notification.platform.includes(getPathname())) {
+                if (notification.is_disposable) {
+                    this.removeNotificationMessage({
+                        key: notification.key,
+                        should_show_again: notification.should_show_again,
+                    });
+                    this.removeNotificationByKey({ key: notification.key });
+                }
+            }
+            return false;
+        });
+    }
+
+    @action.bound
+    handleClientNotifications() {
```

Comment:
I think we should break this up. Break the handling depending on which package is calling it and update the function call at each. Example:

handleAccountNotifications() -> handles the poi/poa/account related stuff
handleCashierNotifications() -> handle cashier notifications

That way we get a cleaner separation and performance wise this will run only certain checks not all of the unrelated checks.

Reply:
Agree!

Reply:
@yashim-deriv @matin-binary In the current implementation, we call `handleClientNotifications()` just once on the first application initialization and only in cases when we want to `refreshNotifications() `e.g. after we request the updated account status from back-end, in `Trader`/`Account` package within 5 components in order to refresh all notifications regardless of their content. We don’t call `handleClientNotifications()` from Cashier at all, and many cashier-related notifications depend on the account state. For this reason, I believe we don't have to split `handleClientNotifications()` into smaller methods. What do you think? 😊

Reply:
We can leave this for now to avoid this PR from getting too large to review. The` handleClientNotifications()` need to be break up imho. Either we change the way of referencing store/ we change the way we do checks or we change the way we update these notifications. Architecture wise if a function becomes too huge or sequence matters too much (like in this case) it is always a sign of an impending doom 😆  For now the splitting of notifications to its own store is good enough. We probably can move this on ~ 👍🏼